modules/sdk-core/src/bitgo/wallet/wallets.ts

```diff
@@ -168,6 +169,65 @@ export class Wallets implements IWallets {
     };
   }
 
+  private async generateLightningWallet(params: GenerateWalletOptions): Promise<LightningWalletWithKeychains> {
+    const reqId = new RequestTracer();
+    this.bitgo.setRequestTracer(reqId);
+
+    const lightningParams = _.pick(params, ['label', 'passphrase', 'enterprise', 'passcodeEncryptionCode']);
+    const { label, passphrase, enterprise, passcodeEncryptionCode } = lightningParams;
+
+    assert(label && typeof label === 'string', 'label parameter is required and must be a string');
+    assert(passphrase && typeof passphrase === 'string', 'passphrase parameter is required and must be a string');
+    assert(enterprise && typeof enterprise === 'string', 'enterprise parameter is required and must be a string');
+    assert(
+      passcodeEncryptionCode && typeof passcodeEncryptionCode === 'string',
+      'passcodeEncryptionCode parameter is required and must be a string'
```

Comment:
Why do we need all these type assertions still? 

AFAIK a lot of the existing assertions like this where just carry over from the JS sdk.

Reply:
Lightning wallet creation is primarily will be done by Express users.

`generateWallet` is also called from BitGo Express which does not pass do any type assertion from its HTTP request. 

So TS type safety is unreliable in that case. Thats why run time assertion is done.

Reply:
If the intention is to add type safety for express users, then create a codec for the request that can be used by the express route. 

The type extracted from the codec can be used downstream. Having this parsing is redundant downstream in the sdk, it only really makes sense upstream in the express controllers.

Reply:
Makes sense. Done.